src/providers/supabase/authProvider.ts

```diff
@@ -1,39 +1,28 @@
 import { supabaseAuthProvider } from 'ra-supabase';
 import { AuthProvider } from 'react-admin';
 import { supabase } from './supabase';
+import { canAccess } from '../commons/canAccess';
 
 const baseAuthProvider = supabaseAuthProvider(supabase, {
-    getIdentity: async user => {
-        const { data, error } = await supabase
-            .from('sales')
-            .select('id, first_name, last_name, avatar')
-            .match({ user_id: user.id })
-            .single();
+    getIdentity: async () => {
+        const sale = getSaleFromLocalStorage();
```

Comment:
I know I'm the one who asked to cache the connected `sale` to avoid multiple calls to supabase when calling `canAccess`, but this new implementation causes an issue: when you invite a new user, click on the invitation link in the email, and fill in your password, then you are still connected as the admin you were before, and not as the newly created sales.

Maybe the easiest fix is to use a local variable instead of the localStorage?

Reply:
fixed